src/query_plan/operation.rs

```diff
@@ -0,0 +1,426 @@
+use apollo_compiler::ast::{Argument, DirectiveList, FieldDefinition, Name, NamedType};
+use apollo_compiler::executable::{
+    Field, Fragment, InlineFragment, Operation, Selection, SelectionSet,
+};
+use apollo_compiler::{Node, Schema};
+use indexmap::IndexMap;
+
+// copy of apollo compiler types that store selections in a map so we can normalize it efficiently
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct NormalizedSelectionSet {
+    pub ty: NamedType,
+    pub selections: IndexMap<String, NormalizedSelection>,
```

Comment:
The meaning of this string as map key isn’t obvious without reading the two `*_key` functions below that create it. PR https://github.com/apollographql/federation-next/pull/95 merges into this PR to use an enum instead.